support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerListener.java

```diff
@@ -29,4 +35,24 @@ public void onDeleted(Run<?,?> run) {
         }
     }
 
+    @Extension
+    public static final class Listener extends RunListener<Run<?, ?>> {
+        @Override
+        public void onStarted(Run<?, ?> run, TaskListener listener) {
+            Cause.UpstreamCause upstreamCause = run.getCause(Cause.UpstreamCause.class);
+            if (upstreamCause != null) {
+                Run upStreamRun = upstreamCause.getUpstreamRun();
+                BuildTriggerAction buildTriggerAction = run.getAction(BuildTriggerAction.class);
+                //Only works when upstream is still running
+                if (upStreamRun != null && upStreamRun.isBuilding() && buildTriggerAction != null) {
+                    try {
+                        TaskListener taskListener = buildTriggerAction.getStepContext().get(TaskListener.class);
+                        taskListener.getLogger().println("Starting building project: " + HyperlinkNote.encodeTo('/' + run.getUrl(), run.getFullDisplayName()));
+                    } catch (Exception e) {
```

Comment:
It not allowed but catches, any operation on listener may end with IOException, so handling it in implementation doesn't look like a standard case. My idea that onStarted should allow throwing exceptions and handle it outside implementation by providing normal logging/description to listener/logger.